src/lib/support/StringBuilder.h

```diff
@@ -89,15 +99,52 @@ class StringBuilderBase
     }
 };
 
-/// a preallocated sized string builder
+/// A preallocated sized string builder
 template <size_t kSize>
 class StringBuilder : public StringBuilderBase
 {
 public:
     StringBuilder() : StringBuilderBase(mBuffer, kSize) {}
 
+    StringBuilder(const char * data, size_t size, bool add_marker_if_overflow = true) : StringBuilder()
+    {
+        AddFormat("%.*s", static_cast<int>(size), data);
```

Comment:
```suggestion
        mWriter.Put(data, size);
        NullTerminate();
```

Reply:
also consider replacing all textual `\0` with '.' so that we can print out odd (invalid) charspans as well. Will make finding bugs easier.